### PR TITLE
Add shrink_iri function

### PR DESCRIPTION
**Pull request by phillord**
Something like this perhaps (will add documentation if you are broadly happy).

I am not sure about the handling of the default case, because I am returning "" when I really mean `None`. It's not easy to fit this into the `Curie` interface. The alternative is just to return a string.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -291,7 +309,7 @@ impl PrefixMapping {
 /// ```
 ///
 /// [`PrefixMapping`]: struct.PrefixMapping.html
-#[derive(Debug)]
+#[derive(Eq,PartialEq,Debug)]
```
Comment (waywardmonkeys) [inline src/lib.rs]: Alphabetize the derives, please.

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
What is supposed to happen if there are multiple matching prefixes?

And yeah, we will need to fix Curie to handle default vs empty prefix... but that can be a separate change.

Adding docs would pretty much finish this up!

Could you also run rustfmt on this? (if not, I can after it lands...)

**Comment by phillord** [comment]
Your question about multiple matching prefixes rather stumped me. As far as I can tell, it doesn't say in the spec. The current solution is arbitrary, depending on `HashMap` iterator order.

Think I have done the rest -- didn't know about `rustfmt` so I've learned something there.

**Review by waywardmonkeys — APPROVED** [review]

**Comment by waywardmonkeys** [comment]
Thanks! I've sort of been on holiday lately ... but this looks good and I'll land it now.

**Comment by waywardmonkeys** [comment]
I've published `0.0.7` with this now.

**Comment by phillord** [comment]
Thanks! No worries about the delay and I hope that you are enjoying the holiday.

**Pull request merged**